### PR TITLE
Module#source_location, Class#source_location

### PR DESCRIPTION
**Pull request by KOBA789**
Run this code (named as `test.rb`):
```
module Hoge
end

class Foobar
end

p Hoge.source_location
p Foobar.source_location
```

You should got:
```
["./test.rb", 1]
["./test.rb", 4]
```

Please some comments.

### CODE REVIEW

**Comment by KOBA789** [comment]
Corresponds with this ticket: https://bugs.ruby-lang.org/issues/13383

File: object.c [hunk 1]
```diff
@@ -2604,6 +2606,19 @@ rb_mod_singleton_p(VALUE klass)
     return Qfalse;
 }
 
+/*
+ * call-seq:
+ *     mod.source_location -> [String, Integer]
+ *
+ * Returns the Ruby source filename and line number defining this module *at first*
+ * or nil if module was not defined in Ruby (i.e. naitve)
```
Comment (willfish) [inline object.c]: *native

**Comment by k0kubun** [comment]
It seems to have a conflict now. Could you rebase this from master?

**Comment by k0kubun** [comment]
Let me close this as it has not been updated for a while. Please reopen this after resolving conflicts. Thanks.

**Pull request closed without merging**